### PR TITLE
Split modules, update readme

### PR DESCRIPTION
**Pull request by ronnyfriedland**
Split core fem and additional exporters into separate directories, compose files and makefiles.

This makes it easier to extend the project in future.

### CODE REVIEW

**Comment by 2Fraennk** [comment]
Got this error when building and starting F.E.M. base:
`
⠧ Container fem_prometheus           Starting                                                                                       0.7s
Error response from daemon: failed to create shim task: OCI runtime create failed: runc create failed: unable to start container process: error during container init: error mounting "/opt/docker/falcon-eye-monitoring/target/config/prometheus.yml" to rootfs at "/etc/prometheus/prometheus.yml": mount /opt/docker/falcon-eye-monitoring/target/config/prometheus.yml:/etc/prometheus/prometheus.yml (via /proc/self/fd/6), flags: 0x5001: not a directory: unknown: Are you trying to mount a directory onto a file (or vice-versa)? Check if the specified host path exists and is the expected type
make: *** [Makefile.fem:19: start] Error 1
`

Suggest try to change the path inside the volume mount of prometheus docker-compose like this:
"
volumes:
      - $PWD/target/fem/config/prometheus.yml:/etc/prometheus/prometheus.yml:ro
      - "

**Comment by 2Fraennk** [comment]
Got this error when building the fem-demoapp:
`
make --file Makefile.fem-demoapp start
docker-compose --file target/demoapp/docker-compose-fem-demoapp.yml up -d
ERROR: .NotADirectoryError: [Errno 20] Not a directory: './target/demoapp/docker-compose-fem-demoapp.yml'
make: *** [Makefile.fem-demoapp:11: start] Error 1
`

**Comment by ronnyfriedland** [comment]
Fixed paths for FEM (core) .

**Comment by ronnyfriedland** [comment]
Cannot reproduce demoapp problem

**Comment by 2Fraennk** [comment]
Container for tel-collector doesn't start: config -files were not found:
```fem_otel-collector:
    image: otel/opentelemetry-collector-contrib:0.63.1
    container_name: fem_otel-collector
    command: ["--config=/etc/otel-collector-config.yml"]
    volumes:
      - $PWD/target/config/otel-collector-fem-config.yml:/etc/otel-collector-config.yml:ro
      - $PWD/target/certs:/tmp/certs:ro

origin-path should be: 
- $PWD/target/fem/config/otel-collector-fem-config.yml:/etc/otel-collector-config.yml:ro
- $PWD/target/fem/certs:/tmp/certs:ro

**Comment by 2Fraennk** [comment]
Container for loki doesn't start: config -files is not found:

```fem_loki:
    image: grafana/loki:main
    container_name: fem_loki
    command: -config.file=/etc/loki/config.yml
    volumes:
      - $PWD/target/config/loki-config.yml:/etc/loki/config.yml:ro

origin path should be:  - $PWD/target/fem/config/loki-config.yml:/etc/loki/config.yml:ro

**Comment by 2Fraennk** [comment]
Plz add "Configuration / Preparation" for the make-config-step as headline to the Readme.md to the section "DB Exporters", like: https://github.com/T-Systems-MMS/falcon-eye-monitoring#configuration--preparation

**Review by 2Fraennk — COMMENTED** [review]
After building the image for demoapp, it's name is "demoapp_fem_demoapp", see:
"CONTAINER ID   IMAGE                                               COMMAND                  CREATED              STATUS                          PORTS                                                                    NAMES
2dd6118e53c6   demoapp_fem_demoapp                                 "/usr/local/bin/mvn-…"   About a minute ago   Up About a minute               0.0.0.0:38080->8080/tcp, :::38080->8080/tcp                              fem_demoapp"
double mention of demoapp?

**Comment by 2Fraennk** [comment]
> Cannot reproduce demoapp problem

Same for me, it works now.

**Review by 2Fraennk — CHANGES_REQUESTED** [review]
"@cp shellexporter/docker-compose-fem-shellexporter.yml target" should be:
"@cp shellexporter/docker-compose-fem-shellexporter.yml target/shellexporter"

**Review by 2Fraennk — CHANGES_REQUESTED** [review]
Makefile.fem-shellexporter should looks like that?
"SHELL := /bin/bash

prepare-fem-shellexporter:
        @cp -rn shellexporter target
        @cp shellexporter/docker-compose-fem-shellexporter.yml target/shellexporter
..."

**Review by 2Fraennk — CHANGES_REQUESTED** [review]
"docker-compose-fem-shellexporter.yml" should change it's context like this?
"...
context: .
..."

File: Makefile.fem [hunk 1]
```diff
@@ -1,35 +1,26 @@
...
-	@cp -n docker-compose-fem-demoapp.yml target
+certs: prepare-fem
+	@openssl req -newkey rsa:4096 -nodes -keyout target/fem/certs/server.key -subj "/C=DE/ST=Sachsen/L=Dresden/O=T-Systems MMS GmbH/CN=monitoring-host" -out target/fem/certs/server.csr
+	@openssl x509 -req -extfile <(printf "subjectAltName=DNS:fem_otel-collector,DNS:localhost,IP:127.0.0.1") -days 365 -in target/fem/certs/server.csr -signkey target/fem/certs/server.key -out target/fem/certs/server.crt
+	@chmod 644 target/fem/certs/server.key target/fem/certs/server.crt
 
-config: target
-	@cp -r config target/
+start: config certs
+	docker-compose --file target/fem/docker-compose-fem.yml up -d
```
Comment (2Fraennk) [inline Makefile.fem:20]: @ronnyfriedland , I think the definition of the project name is needed now, e.g. add: "--project-name fem".
With the separation into modules it must be ensure that all components run in the same network.
As you know without configure the project name, the name for the container as well as the name for the network were created from the directory name by default. That results into different subnets for the containers. The avoid that we have to set the project name and remove the "fem_"-prefix in all docker-compose files.

File: Makefile.fem-dbexporter [hunk 2]
```diff
@@ -1,44 +1,52 @@
...
-start: docker-fem-dbexporter
-	docker-compose --file target/docker-compose-fem-dbexporter.yml up -d
+stop-mongodb: prepare-fem-dbexporter-mongodb
+	docker-compose --file target/dbexporter/docker-compose-fem-dbexporter-mongodb.yml down
 
-stop: docker-fem-dbexporter
-	docker-compose --file target/docker-compose-fem-dbexporter.yml down
+start-postgresql: prepare-fem-dbexporter-postgresql
+	@echo "Please ensure to call 'config-postgresql' before starting"
+	docker-compose --file target/dbexporter/docker-compose-fem-dbexporter-postgresql.yml up -d
```
Comment (2Fraennk) [inline Makefile.fem-dbexporter:46]: @ronnyfriedland , I think the definition of the project name is needed now, e.g. add: "--project-name fem".
With the separation into modules it must be ensure that all components run in the same network.
As you know without configure the project name, the name for the container as well as the name for the network were created from the directory name by default. That results into different subnets for the containers. The avoid that we have to set the project name and remove the "fem_"-prefix in all docker-compose files.

File: Makefile.fem-dbexporter [hunk 3]
```diff
@@ -1,44 +1,52 @@
...
 
 	@echo "postgresql host list (format: host:port[,host:port]): "; \
 	read HOSTS; \
-	sed -i "s/{postgresql_hosts}/$$HOSTS/" target/docker-compose-fem-dbexporter.yml
+	sed -i "s/{postgresql_hosts}/$$HOSTS/" target/dbexporter/docker-compose-fem-dbexporter-postgresql.yml
 
-config: mongodb postgresql
+start-mongodb: prepare-fem-dbexporter-mongodb
+	@echo "Please ensure to call 'config-mongodb' before starting"
+	docker-compose --file target/dbexporter/docker-compose-fem-dbexporter-mongodb.yml up -d
```
Comment (2Fraennk) [inline Makefile.fem-dbexporter:39]: @ronnyfriedland , I think the definition of the project name is needed now, e.g. add: "--project-name fem".
With the separation into modules it must be ensure that all components run in the same network.
As you know without configure the project name, the name for the container as well as the name for the network were created from the directory name by default. That results into different subnets for the containers. The avoid that we have to set the project name and remove the "fem_"-prefix in all docker-compose files.

File: Makefile.fem-demoapp [hunk 4]
```diff
@@ -0,0 +1,14 @@
+SHELL := /bin/bash
+
+prepare-fem-demoapp:
+	@mkdir -p target/demoapp
+	@cp -rn demoapp target/
+
+start: prepare-fem-demoapp
+	docker-compose --file target/demoapp/docker-compose-fem-demoapp.yml up -d
```
Comment (2Fraennk) [inline Makefile.fem-demoapp:8]: @ronnyfriedland , I think the definition of the project name is needed now, e.g. add: "--project-name fem".
With the separation into modules it must be ensure that all components run in the same network.
As you know without configure the project name, the name for the container as well as the name for the network were created from the directory name by default. That results into different subnets for the containers. The avoid that we have to set the project name and remove the "fem_"-prefix in all docker-compose files.

File: Makefile.fem-shellexporter [hunk 5]
```diff
@@ -1,17 +1,14 @@
...
-docker-fem-shellexporter: target
-	@cp docker-compose-fem-shellexporter.yml target
+prepare-fem-shellexporter:
+	@mkdir -p target/shellexporter
 	@cp -rn shellexporter target
 
-start: docker-fem-shellexporter
-	docker-compose --file target/docker-compose-fem-shellexporter.yml up -d
+start: prepare-fem-shellexporter
+	docker-compose --file target/shellexporter/docker-compose-fem-shellexporter.yml up -d
```
Comment (2Fraennk) [inline Makefile.fem-shellexporter:8]: @ronnyfriedland , I think the definition of the project name is needed now, e.g. add: "--project-name fem".
With the separation into modules it must be ensure that all components run in the same network.
As you know without configure the project name, the name for the container as well as the name for the network were created from the directory name by default. That results into different subnets for the containers. The avoid that we have to set the project name and remove the "fem_"-prefix in all docker-compose files.

File: dbexporter/docker-compose-fem-dbexporter-mongodb.yml [hunk 6]
```diff
@@ -21,13 +21,3 @@ services:
     ]
     networks:
       - fem_monitoring
```
Comment (2Fraennk) [inline dbexporter/docker-compose-fem-dbexporter-mongodb.yml:23]: @ronnyfriedland , I think the definition of the project name is needed now, e.g. add: "--project-name fem".
With the separation into modules it must be ensure that all components run in the same network.
As you know without configure the project name, the name for the container as well as the name for the network were created from the directory name by default. That results into different subnets for the containers. The avoid that we have to set the project name and remove the "fem_"-prefix in all docker-compose files.

File: dbexporter/docker-compose-fem-dbexporter-mongodb.yml [hunk 7]
```diff
@@ -21,13 +21,3 @@ services:
...
-  fem_postgresql-exporter:
-    image: prometheuscommunity/postgres-exporter:v0.11.1
-    restart: unless-stopped
-    container_name: fem_postgresql_exporter
-    ports:
-      - "29187:9187"
-    environment:
-      DATA_SOURCE_NAME: "postgresql://{postgresql_username}:{postgresql_password}@{postgresql_hosts}/postgres?sslmode=require"
-    networks:
```
Comment (2Fraennk) [inline dbexporter/docker-compose-fem-dbexporter-mongodb.yml:32]: @ronnyfriedland , I think the definition of the project name is needed now, e.g. add: "--project-name fem".
With the separation into modules it must be ensure that all components run in the same network.
As you know without configure the project name, the name for the container as well as the name for the network were created from the directory name by default. That results into different subnets for the containers. The avoid that we have to set the project name and remove the "fem_"-prefix in all docker-compose files.

File: dbexporter/docker-compose-fem-dbexporter-postgresql.yml [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+  fem_postgresql-exporter:
+    image: prometheuscommunity/postgres-exporter:v0.11.1
+    restart: unless-stopped
+    container_name: fem_postgresql_exporter
+    ports:
+      - "29187:9187"
+    environment:
+      DATA_SOURCE_NAME: "postgresql://{postgresql_username}:{postgresql_password}@{postgresql_hosts}/postgres?sslmode=require"
+    networks:
+      - fem_monitoring
```
Comment (2Fraennk) [inline dbexporter/docker-compose-fem-dbexporter-postgresql.yml:16]: @ronnyfriedland , I think the definition of the project name is needed now, e.g. add: "--project-name fem".
With the separation into modules it must be ensure that all components run in the same network.
As you know without configure the project name, the name for the container as well as the name for the network were created from the directory name by default. That results into different subnets for the containers. The avoid that we have to set the project name and remove the "fem_"-prefix in all docker-compose files.

File: dbexporter/docker-compose-fem-dbexporter-postgresql.yml [hunk 9]
```diff
@@ -0,0 +1,16 @@
+version: "3.8"
+
+networks:
+  fem_monitoring:
```
Comment (2Fraennk) [inline dbexporter/docker-compose-fem-dbexporter-postgresql.yml:4]: @ronnyfriedland , I think the definition of the project name is needed now, e.g. add: "--project-name fem".
With the separation into modules it must be ensure that all components run in the same network.
As you know without configure the project name, the name for the container as well as the name for the network were created from the directory name by default. That results into different subnets for the containers. The avoid that we have to set the project name and remove the "fem_"-prefix in all docker-compose files.

File: demoapp/docker-compose-fem-demoapp.yml [hunk 10]
```diff
@@ -6,7 +6,8 @@ networks:
 services:
   fem_demoapp:
```
Comment (2Fraennk) [inline demoapp/docker-compose-fem-demoapp.yml:7]: @ronnyfriedland , I think the definition of the project name is needed now, e.g. add: "--project-name fem".
With the separation into modules it must be ensure that all components run in the same network.
As you know without configure the project name, the name for the container as well as the name for the network were created from the directory name by default. That results into different subnets for the containers. The avoid that we have to set the project name and remove the "fem_"-prefix in all docker-compose files.

File: fem/docker-compose-fem.yml [hunk 11]
```diff
@@ -77,45 +77,44 @@ services:
     container_name: fem_opensearch_exporter
     command:
      - '--es.uri=http://fem_opensearch:9200'
-    restart: unless-stopped
     networks:
       - fem_monitoring
```
Comment (2Fraennk) [inline fem/docker-compose-fem.yml:81]: @ronnyfriedland , I think the definition of the project name is needed now, e.g. add: "--project-name fem".
With the separation into modules it must be ensure that all components run in the same network.
As you know without configure the project name, the name for the container as well as the name for the network were created from the directory name by default. That results into different subnets for the containers. The avoid that we have to set the project name and remove the "fem_"-prefix in all docker-compose files.

File: fem/docker-compose-fem.yml [hunk 12]
```diff
@@ -77,45 +77,44 @@ services:
...
-    image: jaegertracing/jaeger-es-index-cleaner:1.39
+    image: jaegertracing/jaeger-es-index-cleaner:1.42
     container_name: fem_opensearch-cleaner
     command: [
       "7",
       "http://fem_opensearch:9200"
     ]
     networks:
-    - fem_monitoring
+      - fem_monitoring
```
Comment (2Fraennk) [inline fem/docker-compose-fem.yml:92]: @ronnyfriedland , I think the definition of the project name is needed now, e.g. add: "--project-name fem".
With the separation into modules it must be ensure that all components run in the same network.
As you know without configure the project name, the name for the container as well as the name for the network were created from the directory name by default. That results into different subnets for the containers. The avoid that we have to set the project name and remove the "fem_"-prefix in all docker-compose files.

File: fem/docker-compose-fem.yml [hunk 13]
```diff
@@ -77,45 +77,44 @@ services:
...
     restart: unless-stopped
     container_name: fem_prometheus
     ports:
       - "29090:9090" #ui, notls
     volumes:
-      - $PWD/target/config/prometheus.yml:/etc/prometheus/prometheus.yml:ro
+      - $PWD/target/fem/config/prometheus.yml:/etc/prometheus/prometheus.yml:ro
       - fem_prometheus_data:/prometheus
     networks:
       - fem_monitoring
```
Comment (2Fraennk) [inline fem/docker-compose-fem.yml:105]: @ronnyfriedland , I think the definition of the project name is needed now, e.g. add: "--project-name fem".
With the separation into modules it must be ensure that all components run in the same network.
As you know without configure the project name, the name for the container as well as the name for the network were created from the directory name by default. That results into different subnets for the containers. The avoid that we have to set the project name and remove the "fem_"-prefix in all docker-compose files.

File: fem/docker-compose-fem.yml [hunk 14]
```diff
@@ -77,45 +77,44 @@ services:
...
     ports:
       - "23000:3000" #ui, notls
     volumes:
       - fem_grafana_data:/var/lib/grafana
-      - $PWD/target/config/grafana-dashboard-templates:/var/lib/grafana-dashboard-templates
-      - $PWD/target/config/grafana-provisioning:/etc/grafana/provisioning
+      - $PWD/target/fem/config/grafana-dashboard-templates:/var/lib/grafana-dashboard-templates
+      - $PWD/target/fem/config/grafana-provisioning:/etc/grafana/provisioning
     networks:
       - fem_monitoring
```
Comment (2Fraennk) [inline fem/docker-compose-fem.yml:119]: @ronnyfriedland , I think the definition of the project name is needed now, e.g. add: "--project-name fem".
With the separation into modules it must be ensure that all components run in the same network.
As you know without configure the project name, the name for the container as well as the name for the network were created from the directory name by default. That results into different subnets for the containers. The avoid that we have to set the project name and remove the "fem_"-prefix in all docker-compose files.

File: fem/docker-compose-fem.yml [hunk 15]
```diff
@@ -141,14 +140,14 @@ services:
     container_name: fem_loki
     command: -config.file=/etc/loki/config.yml
     volumes:
-      - $PWD/target/config/loki-config.yml:/etc/loki/config.yml:ro
+      - $PWD/target/fem/config/loki-config.yml:/etc/loki/config.yml:ro
       - fem_loki_data:/tmp
     ports:
       - 23100:3100 #api, notls
     networks:
       - fem_monitoring
```
Comment (2Fraennk) [inline fem/docker-compose-fem.yml:148]: @ronnyfriedland , I think the definition of the project name is needed now, e.g. add: "--project-name fem".
With the separation into modules it must be ensure that all components run in the same network.
As you know without configure the project name, the name for the container as well as the name for the network were created from the directory name by default. That results into different subnets for the containers. The avoid that we have to set the project name and remove the "fem_"-prefix in all docker-compose files.

**Review by 2Fraennk — CHANGES_REQUESTED** [review]
@ronnyfriedland , I think the definition of the project name is needed now, e.g. add: "--project-name fem".
With the separation into modules it must be ensure that all components run in the same network.
As you know without configure the project name, the name for the container as well as the name for the network were created from the directory name by default. That results into different subnets for the containers. The avoid that we have to set the project name and remove the "fem_"-prefix in all docker-compose files.

**Review by 2Fraennk — CHANGES_REQUESTED** [review]
just one more to go  ;)
https://github.com/T-Systems-MMS/falcon-eye-monitoring/pull/16/files#r1105514012

**Comment by ronnyfriedland** [comment]
added missing project parameter to mongodb exporter

**Review by 2Fraennk — APPROVED** [review]

**Pull request merged**